consensus/istanbul/backend/backend.go

```diff
@@ -430,21 +432,46 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, aggregatedSeal types.Istan
 	})
 
 	sb.logger.Info("Committed", "address", sb.Address(), "round", aggregatedSeal.Round.Uint64(), "hash", proposal.Hash(), "number", proposal.Number().Uint64())
-	// - if the proposed and committed blocks are the same, send the proposed hash
-	//   to commit channel, which is being watched inside the engine.Seal() function.
-	// - otherwise, we try to insert the block.
-	// -- if success, the ChainHeadEvent event will be broadcasted, try to build
-	//    the next block and the previous Seal() will be stopped.
-	// -- otherwise, a error will be returned and a round change event will be fired.
-	if sb.proposedBlockHash == block.Hash() {
-		// feed block hash to Seal() and wait the Seal() result
-		sb.commitCh <- block
+
+	sealHash := sb.SealHash(block.Header())
```

Comment:
I think the `sealhash` calculated here will be different than when it's calculated when the cache entry is saved.

Reply:
`sealHash` will only calculate the hash based on block header without taking `seal` into account. 
And epoch snark data is store outside of header.
So sealHash makes sense here.